scrapy/extensions/checksettings.py

```diff
@@ -0,0 +1,54 @@
+"""
+Check settings extension
+
+See documentation in docs/topics/extensions.rst
+"""
+import logging
+import pprint
+import operator
+
+from fuzzywuzzy.fuzz import ratio
+from scrapy import signals
+from scrapy.exceptions import NotConfigured
+
+logger = logging.getLogger(__name__)
+
+
+class CheckSettings:
+    def __init__(self, crawler):
+        self.settings = crawler.settings
+        self.check_settings_min = crawler.settings.getint("CHECK_SETTINGS_MIN")
```

Comment:
On the topic of settings for this extensions, I wonder if it would not make sense to provide a setting do indicate a list of settings that can be ignored by this extension, so that they are considered used even if they are not.